app/models/check_flag_job.rb

```diff
@@ -0,0 +1,51 @@
+class CheckFlagJob < Struct.new(:flag, :token)
+
+  def perform
+    flag.access_token = token
+    flag.resolved?
+  end
+
+  def success(job)
+    flag.job_id = nil
+    flag.save
+  end
+
+  def error(job, exception)
+    @exception = exception
+  end
+
+  def failure(job)
+    flag.job_id = nil
+    flag.status = "failed"
+    flag.save
+  end
+
+  def max_attempts
+    # Retries happen at 5 + n^4 seconds, where n = number of attempts
+    # 12 gives last retry at 6 hours
```

Comment:
this comment is out of date, since reschedule_at() overrides the logic. consider removing.